src/cargo/ops/cargo_add/crate_spec.rs

```diff
@@ -28,7 +28,24 @@ impl CrateSpec {
             .map(|(n, v)| (n, Some(v)))
             .unwrap_or((pkg_id, None));
 
-        PackageName::new(name)?;
+        let package_name = PackageName::new(name);
+        if !pkg_id.contains("@") && package_name.is_err() {
+            for (idx, ch) in pkg_id.char_indices() {
+                if !(unicode_xid::UnicodeXID::is_xid_continue(ch) || ch == '-') {
+                    let mut suggested_pkg_id = pkg_id.to_string();
+                    String::insert_str(&mut suggested_pkg_id, idx, "@");
+                    if let Ok(_) = CrateSpec::resolve(&suggested_pkg_id.as_str()) {
+                        return Err(anyhow::anyhow!(format!(
+                            "Possible missing `@` detected in package name `{}`. Try `{}` instead.",
```

Comment:
By convention, the first letter in the errro message should be lower case.

Reply:
Done